### ISSUE TITLE
Issues with ConfigurationManager

### ISSUE THREAD

**Issue by birojnayak** [CoreWCF/CoreWCF#445]
1)When the service tag missing name, the service is ignored. @mconnew looking at WCF and suggest what should be alternative (generate unique id and add or others).
2)When the Service and Contract are coming from a reference project or assembly, the service and contract type are not resolved to appropriate type.

**Comment by mconnew** [comment]
I took a look at the NetFx code and how ServiceEndpointElement.Name is used and Name wasn't meant to be a unique value. A valid snippet of config is the following:
```xml
        <endpoint address="" binding="basicHttpBinding" contract="ICalculator"/>
        <endpoint address="mex" binding="mexHttpBinding" contract="IMetadataExchange"/>
```
They both have a null name, but are obviously different endpoints. We need to always generate an artificial name to use as the key in the dictionary and not use the name field of the endpoint. We can't key on just the address because different transports have different scopes, so the same address can be used for NetTcp and Http without any overlap. I believe we should just be able to use the address, a delimeter, and the binding name concatenated together. That simple change in what key is used at the end of ConfigurationHolder.AddServiceEndpoint should fix the issue for ServiceEndpoint. I'm going to check what to do about \<service\> but wanted to update you with what I have worked out so far.
